src/shell/Models/ShellExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace NeoShell.Models
+{
+    class ShellExtensions : List<ShellExtension>
+    {
+        public bool TryFindCommand(string[] args, out ShellExtension? command)
```

Comment:
minor nit: out param should be marked `[MaybeNullWhen(false)]` and *not* have the `?` marker

Reply:
good to know, thx